Sources/StreamChat/Database/DTOs/MessageReactionDTO.swift

```diff
@@ -8,23 +8,34 @@ import Foundation
 @objc(MessageReactionDTO)
 final class MessageReactionDTO: NSManagedObject {
     @NSManaged fileprivate var id: String
-    
+
+    // holds the rawValue of LocalReactionState
+    @NSManaged fileprivate var localStateRaw: String?
     @NSManaged var type: String
     @NSManaged var score: Int64
-    @NSManaged var createdAt: Date
-    @NSManaged var updatedAt: Date
+    @NSManaged var createdAt: Date?
+    @NSManaged var updatedAt: Date?
     @NSManaged var extraData: Data
     
     @NSManaged var message: MessageDTO
     @NSManaged var user: UserDTO
-    
+
+    // internal field needed to sync data optimistically
+    @NSManaged var version: String?
```

Comment:
Can you be more descriptive on why we need this version prop? Didn't understand yet why need it 🤔

Reply:
this is avoid one easy out-of-sync scenario with the retry error handler, when adding a reaction fails the reaction is only updated unless it was changed in the meantime

Reply:
Also didn't get this one, sorry 😄 Could you explain again what is the scenario we're trying to prevent having the `version`?

Reply:
this is used to perform optimistic locking style updates (general idea is linked [here](https://en.wikipedia.org/wiki/Optimistic_concurrency_control))

example:

- insert reaction Z with version X
- [ other things happen and perhaps alter reaction Z, in that case they will erase / overwrite the version field ]
- update reaction Z if version is X